### PR TITLE
Delta Diff: Compare Delta Tables button

### LINKED ISSUE #4949: Delta Diff: "Compare Delta Lake Tables" experimental button
Create a "Compare Delta Tables" experimental button that will, at first, show "under construction", and further on will be the gateway to the Delta diff functionality.

### PR DESCRIPTION
**Pull request by Jonathan-Rosenberg**
This PR includes a button to compare Delta Lake tables. Currently, this feature is under construction so it only shows a "coming soon" modal.
Screenshot 2023-01-09 at 18 52 09
Screenshot 2023-01-09 at 18 52 17
Screenshot 2023-01-09 at 18 52 30

Closes #4949 

(It also includes some formatting... This is the reason for the number of changes in the `webui/src/lib/api/index.js` file)

### CODE REVIEW

File: webui/src/lib/components/controls.jsx [hunk 1]
```diff
@@ -406,3 +406,20 @@ export const ExitConfirmationDialog = ({dialogAlert, dialogDescription, onExit,
         </Dialog>
     )
 };
+
+
+export const ExperimentalOverlayTooltip = ({children, show = true, placement="auto"}) => {
+    const experimentalTooltip = () => (
+        <Tooltip id="button-tooltip" >
+            Experimental
```
Comment (johnnyaug) [inline webui/src/lib/components/controls.jsx:414]: I would wait with the "experimental" label until the actual implementation.

File: webui/src/pages/repositories/repository/compare.jsx [hunk 2]
```diff
@@ -73,61 +89,82 @@ const CompareList = ({ repo, reference, compareReference, prefix, onSelectRef, o
...
-                                        if (compareReference) q.compare = compareReference.id;
-                                        if (reference) q.ref = reference.id;
-                                        return {
-                                            pathname: '/repositories/:repoId/compare',
-                                            params: {repoId: repo.id},
-                                            query: q
+                    <>
+                        <ComingSoonModal display={showComingSoonModal}
+                                         onCancel={() => setShowComingSoonModal(false)}>
+                            <div>lakeFS Delta Lake tables diff is under development</div>
```
Comment (johnnyaug) [inline webui/src/pages/repositories/repository/compare.jsx:95]:
```suggestion
                            <div>We are still working on this feature. Check back soon.</div>
```

**Review by johnnyaug — APPROVED** [review]
Cooolll.
See minor comments.

File: webui/src/pages/repositories/repository/compare.jsx [hunk 3]
```diff
@@ -73,61 +89,82 @@ const CompareList = ({ repo, reference, compareReference, prefix, onSelectRef, o
...
+                                                };
+                                            }}/>
+                                    )}
+                                </span>
+                            </Card.Header>
+                            <Card.Body>
+                                <Table borderless size="sm">
+                                    <tbody>
+                                    {results.map(entry => {
+                                        if(!showDeltaDiffButton && entry.path.includes("_delta_log")) {
```
Comment (talSofer) [inline webui/src/pages/repositories/repository/compare.jsx:139]: This won't work if the delta table is in from a second order. 
See the example below - 
Screen Shot 2023-01-10 at 10 25 29

I think that we should always show the button, and in the delta comparison view add the logic that only displays delta tables.

**Review by talSofer — CHANGES_REQUESTED** [review]
Very cool! 
Found a bug in the button display logic

**Review by talSofer — APPROVED** [review]
Thanks!

File: webui/src/lib/api/index.js [hunk 4]
```diff
@@ -969,14 +982,29 @@ class Templates {
...
+            {method: 'GET'});
         if (!response.ok) {
             throw new Error(await extractError(response));
         }
         return response.text();
     }
 }
 
+class Statistics {
+    async sendStats(statsEvents) {
```
[the remark this answers is not in the document]
Comment (Jonathan-Rosenberg) [reply webui/src/lib/api/index.js]: done

File: webui/src/pages/repositories/repository/compare.jsx (hunk 3 above)
Comment (Jonathan-Rosenberg) [reply webui/src/pages/repositories/repository/compare.jsx]: done

**Pull request merged**